updating/update-using-custom-machine-config-pools.adoc

```diff
@@ -0,0 +1,167 @@
+[id="update-using-custom-machine-config-pools"]
+= Use of customized machine config pools during update
+include::modules/common-attributes.adoc[]
+:context: update-using-custom-machine-config-pools
+
+toc::[]
+
+In OpenShift V4 nodes are not considered individually. Nodes are considered in groups i.e. machine config pools(MCP). In a typical OCP cluster we have two MCPs. The MCP for the control plane is called master pool and other group which have worker nodes is called worker pool. During an OpenShift update all pools are updated concurrently.The MCP update process involves updating the machines to the new OS version(if required) along with updating some daemonsets and reboot it to comeplete the process. The MCP gives us the ability to pause a pool and it will make sure that the OS is not updated/rebooted during the update process. We can create more than one customized machine config pools out of the the worker pool and will give us a lot of control on the sequence in which the worker nodes are updated.
+
+[NOTE]
+----
+Creating custom MCP using master nodes is not suported.
+Machine config operator (responsible for updating the nodes) will ignore the custom MCP using the master nodes.
+This is required to make sure control plane nodes remain stable.
+----
+
+An OpenShift or Kubernetes cluster is highly available by design and there are many Kuberenetes features (e.g. pod disruption budget, pod affinity, health checks, replicas) which can make the application highly available in case of node failures.
+
+However there might be some secenarios where a controlled rollout of upgrade to the worker nodes is desired. This helps to make sure the mission critical application stays available during the whole update process even if the upgrade process cause a failure. It is also useful in scenarios where we can not have big maintaine window to complete the whole update process. This is can be characterized as a canary release i.e. we can control the rollout of worker nodes by controlling the machine config pools. After the update of the first MCP, the application compatibility can be verfied and then the whole fleet can be updated gradually to the new version.
+
+[NOTE]
+----
+A node can be part of at most one MCP. The machine config operator rolls out update to all the unpaused worker pools simultaneously.
+For example, if maxUnavailable is set to one then 1 node from each MCPs will be updated concurrently.
+
+During the update process of nodes (as per are the  `maxUnavailable` value) in a MCP are cordoned and drained before it is updated.
+Once the update process completes the nodes are un-cordned.
+----
+
+== Workflow of doing a canary release using MCP
```

Comment:
nit: "release" is a bit loaded.  I usually think it means "OpenShift release image (and referenced images)", like `quay.io/openshift-release-dev/ocp-release:4.8.0-x86_64`.  Can we use something like `Updating canary machine config pools`?

Reply:
Updating the heading as `Workflow for updating worker nodes with canary rollout`

Reply:
Fixed in https://github.com/openshift/openshift-docs/commit/c27e14a1b4785cc8704f11d191797f198c10f027